Protacon.RxMq.AzureServiceBusLegacy.Tests/AzureBusTopicIntegrationTests.cs

```diff
@@ -78,8 +80,70 @@ await listener
         }
 
         [Fact]
-        public void WhenMultipleThreadsAreWriting_ThenSubscribingWorksAsExpected()
+        public async void WhenTopicIsRemoved_ThenTopicIsCreatedAgain()
         {
+            var settings = TestSettings.MqSettingsForTopic();
+
+            var subscriber = new AzureBusTopicSubscriber(settings, _ => { }, _ => { });
+            var publisher = new AzureBusTopicPublisher(settings, _ => { }, _ => throw new InvalidOperationException("Errorlogger"));
+
+            var id = Guid.NewGuid();
+            var listener = subscriber.Messages<TestMessageForTopic>();
+
+
+            var name = settings.TopicNameBuilder(typeof(TestMessageForTopic));
+            var nameSpace = NamespaceManager.CreateFromConnectionString(settings.ConnectionString);
+            var topic = await nameSpace.GetTopicAsync(name);
+            topic.Should().NotBeNull("Topic should exist at this point");
+
+            await nameSpace.DeleteTopicAsync(name);
+
+            Thread.Sleep(TimeSpan.FromSeconds(30));
+
+            await publisher.SendAsync(new TestMessageForTopic
+            {
+                ExampleId = id
+            });
+
+            await listener
+                .Where(x => x.ExampleId == id)
+                .Timeout(TimeSpan.FromSeconds(30))
+                .FirstAsync();
+        }
+
+        [Fact]
+        public async void WhenSusbcriptionIsRemoved_ThenSubscriptionIsCreatedAgain()
+        {
+            var settings = TestSettings.MqSettingsForTopic();
+
+            var subscriber = new AzureBusTopicSubscriber(settings, _ => { }, _ => { });
+            var publisher = new AzureBusTopicPublisher(settings, _ => { }, _ => throw new InvalidOperationException("Errorlogger"));
+
+            var id = Guid.NewGuid();
+            var listener = subscriber.Messages<TestMessageForTopic>();
+
+
+            var name = settings.TopicNameBuilder(typeof(TestMessageForTopic));
+            var nameSpace = NamespaceManager.CreateFromConnectionString(settings.ConnectionString);
+            var topic = await nameSpace.GetTopicAsync(name);
+
+            var subscriptions = await nameSpace.GetSubscriptionsAsync(topic.Path);
+            foreach (var subscriptionDescription in subscriptions)
+            {
+                await nameSpace.DeleteSubscriptionAsync(topic.Path, subscriptionDescription.Name);
+            }
+
+            Thread.Sleep(TimeSpan.FromSeconds(30));
```

Comment:
Could this be implemented so that test waits subscription to be deleted? This allows longer timeout (more robust) and in other hand makes test faster in common cases to execute.

Reply:
It could be, but we are not waiting for subscriptions to be deleted. We are waiting for the recreation of those subscriptions.

Problem is that we would need to put the wait to the "listen" part of messaging, but we can only send the message after we are sure that the subscription is created. I chose the lazy way because these tests are not ran by CI.